ios/browser/api/web_view/universal_link_block_util.mm

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#import <Foundation/Foundation.h>
+
+#import "brave/ios/browser/api/web_view/brave_web_view.h"
+#include "ios/web/public/web_state.h"
+#import "ios/web_view/internal/cwv_web_view_internal.h"
+
+namespace brave {
+
+bool ShouldBlockUniversalLinks(web::WebState* webState, NSURLRequest* request) {
```

Comment:
it's also not clear to me how/why we're handling this differently than upstream. Also why is this implemented in `universal_link_block_util.mm` and defined in `wk_navigation_action_policy_util.h`? `ShouldBlockJavaScript` is also defined in `wk_navigation_action_policy_util.h` and implemented in a different file. This also doesn't seem like it belongs in `ios/browser/api`? Isn't that just for things we expose to swift?

Reply:
fixed in https://github.com/brave/brave-core/pull/26957/commits/0bcc0de8685bed80111d84d52c00f00735fdba83